### PR TITLE
Add test for issue #29595

### LINKED ISSUE #29595: ICE: `Unimplemented` when trying to select an implementation for constant trait item reference
Code:

```
#![feature(associated_consts)]

trait Tr {
    const C: Self;
}

fn main() {
    let a: u8 = Tr::C;
}
```

Output:

```
<anon>:4:5: 4:19 error: internal compiler error: Encountered error `Unimplemented` when trying to select an implementation for constant trait item reference.
<anon>:4     const C: Self;
             ^~~~~~~~~~~~~~
note: the compiler unexpectedly panicked. this is a bug.
note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports
note: run with `RUST_BACKTRACE=1` for a backtrace
thread 'rustc' panicked at 'Box<Any>', ../src/libsyntax/diagnostic.rs:176
```

Reproduces on nightly.

### PR DESCRIPTION
**Pull request by mattico**
Closes #29595

Couldn't get this to run locally, all the compile-fail tests are ignored... let's see what Travis says.

### CODE REVIEW

**Comment by mattico** [comment]
CC @petrochenkov

**Comment by estebank** [comment]
@mattico Instead of using a `compile-fail` test, could you move this new test to `ui`?

**Comment by aturon** [comment]
r=me once the test is moved as per @estebank's request.

@bors delegate=estebank

**Comment by mattico** [comment]
@estebank done.

File: src/test/ui/span/issue-29595.stderr [hunk 1]
```diff
@@ -0,0 +1,10 @@
+error[E0277]: the trait bound `u8: Tr` is not satisfied
+  --> src\test\ui\span\issue-29595.rs:18:17
```
Comment (estebank) [inline src/test/ui/span/issue-29595.stderr]: [This should be `$DIR/issue-29595.rs:18:17`](https://travis-ci.org/rust-lang/rust/jobs/210693743#L4363).

**Comment by mattico** [comment]
@estebank fixed now 👍

**Pull request merged**